daemon/src/db/cull_old_dlcs.rs

```diff
@@ -149,16 +104,60 @@ mod tests {
     use time::Duration;
 
     #[tokio::test]
-    async fn given_two_rollovers_when_cull_old_dlcs_then_only_last_one_has_some_dlc() {
+    async fn given_only_contract_setup_completed_and_no_rollover_when_culled_event_has_some_dlc() {
         let db = memory().await.unwrap();
 
-        let (cfd, contract_setup, rollovers) = rolled_over_cfd(NonZeroU8::new(2).unwrap());
-        let most_recent_rollover = rollovers.last().unwrap().event.clone();
-
+        let (cfd, contract_setup) = contract_setup_completed_cfd();
+        let contract_setup_cloned = contract_setup.event.clone();
         db.insert_cfd(&cfd).await.unwrap();
+        db.append_event(contract_setup).await.unwrap();
+
+        db.cull_old_dlcs().await.unwrap();
+
+        let mut conn = db.inner.acquire().await.unwrap();
+        let mut db_tx = conn.begin().await.unwrap();
+        let events = load_cfd_events(&mut db_tx, cfd.id(), 0).await.unwrap();
+        db_tx.commit().await.unwrap();
+
+        assert_eq!(events[0].event, contract_setup_cloned,);
+    }
 
+    #[tokio::test]
+    async fn given_one_rollovers_when_cull_then_first_event_has_none_dlc_and_second_hassome_dlc() {
+        let db = memory().await.unwrap();
+
+        let (cfd, contract_setup, rollovers) = rolled_over_cfd(NonZeroU8::new(1).unwrap());
+        db.insert_cfd(&cfd).await.unwrap();
         db.append_event(contract_setup).await.unwrap();
+        for rollover in rollovers {
+            db.append_event(rollover).await.unwrap();
+        }
+
+        db.cull_old_dlcs().await.unwrap();
+
+        let mut conn = db.inner.acquire().await.unwrap();
+        let mut db_tx = conn.begin().await.unwrap();
+        let events = load_cfd_events(&mut db_tx, cfd.id(), 0).await.unwrap();
+        db_tx.commit().await.unwrap();
+
+        assert!(matches!(
+            &events[0].event,
+            EventKind::ContractSetupCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[1].event,
+            EventKind::RolloverCompleted { dlc: Some(_), .. },
```

Comment:
🔧 Daniel suggested to always check against the original here, like we do in the previous test.